### PR TITLE
Optimal Payment: Add store method

### PR DESCRIPTION
**Pull request by britth**
Add support for store on Optimal Payment, which calls verify with
the cc and any options passed. You can use the authorization value
returned from the successful transaction to issue a purchase or auth,
which will utilize that same payment method.

Remote:
18 tests, 85 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Unit:
19 tests, 75 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

**Comment by curiousepic** [comment]
Weird. This makes me wonder if there's some option we should be setting for normal Verify calls to prevent the data being stored.

**Comment by curiousepic** [comment]
Oh, I would add failed case tests as well, if it makes sense.

File: test/remote/gateways/remote_optimal_payment_test.rb [hunk 1]
```diff
@@ -57,6 +58,52 @@ def test_successful_verify
...
+    response = @gateway.store(@invalid_card, @options)
+    assert_failure response
+    assert_not_nil response.authorization
+    assert_equal 'ERROR', response.params['decision']
+
+
+    assert stored_purchase = @gateway.stored_purchase(@amount, response.authorization)
+    assert_failure stored_purchase
+    assert_equal 'ERROR', stored_purchase.params['decision']
+  end
```
Comment (britth) [inline test/remote/gateways/remote_optimal_payment_test.rb]: @curiousepic how do these last two seem for failed test cases? I don't see a specific invalid test card, so just created a card with a blank PAN for these two

Comment (curiousepic) [reply test/remote/gateways/remote_optimal_payment_test.rb]: 👍

**Pull request merged**